src/Ormolu/Diff/Text.hs

```diff
@@ -8,26 +8,35 @@
 module Ormolu.Diff.Text
   ( TextDiff,
     diffText,
+    selectSpans,
     printTextDiff,
   )
 where
 
 import Control.Monad
 import qualified Data.Algorithm.Diff as D
+import Data.IntSet (IntSet)
+import qualified Data.IntSet as IntSet
+import Data.List (foldl')
 import Data.Maybe (listToMaybe)
 import Data.Text (Text)
 import qualified Data.Text as T
+import GHC.Types.SrcLoc
 import Ormolu.Terminal
 
 ----------------------------------------------------------------------------
 -- Types
 
 -- | Result of diffing two 'Text's.
-data TextDiff = TextDiff FilePath DiffList
+data TextDiff = TextDiff
+  { textDiffPath :: FilePath,
+    textDiffDiffList :: DiffList,
+    textDiffSelectedLines :: IntSet
```

Comment:
Maybe add a comment also here that mentions that an empty `textDiffSelectedLines` means "all lines are selected"?

Reply:
Done!